### DISCUSSION TITLE
Add SKALE Network (Multichain) to Wagmi Chains Package

### DISCUSSION THREAD

**Discussion by TheGreatAxios** [wagmi-dev/references#145] [General]
Hey Wagmi Team, 

I am looking to add SKALE Network support to Wagmi via the @wagmi/chains package. A bit of background about SKALE:

- SKALE is multi-chain. Currently there are 20 Mainnet SKALE Chains
- SKALE has zero-gas fees in the form of sFUEL. sFUEL is the gas token that is distributed by chain owners
- Developers on SKALE may often utilize multiple chains within a dApp

Questions I Have
---
1. What is the architecture that would best suit adding multiple mainnet and testnet chains?
   a. 1 file per each [potentially 20+ files]
   b. 1 file for all of them [larger file]
   c. SKALE Network Folder with listed in the folder
2. Because chains are all apart of the same "network", would it be acceptable to also export a couple of custom objects for example:
   - a **skale** object where you can call skale["chain_name"]
   - skale.hubs which can resolve to an array of specific skale chains

Once there is a bit more clarity on how best to contribute I will add them. 

Thanks team!

**Comment by payvint** [comment]
Just to add to the question 1:
The architecture should be suitable to add new chains on mainnet In a future.

**Reply by TheGreatAxios** [reply]
PR Created: https://github.com/wagmi-dev/references/pull/157

This follows the pattern of all other networks with one minor tweak which is bundling all SKALE Network Chains together into a folder to make it simpler for SKALE Chain Owners to add their own chains in the future.
